src/js/track.js

```diff
@@ -0,0 +1,83 @@
+var _ = require('underscore');
+var d3 = require('d3');
+var $ = require('jquery');
+var D3SVGCellRenderer = require('./d3_svg_cell_renderer');
+var utils = require('./utils');
+
+module.exports = {};
+
+var defaultTrackConfig = {
+	label: 'Gene',
+	id_member: 'sample',
+	cell_height: 20,
+	track_height: 20,
+	track_padding: 2.5,
+}; 
+
+function Track(name, oncoprint, data, config) {
+	var self = this;
+	self.name = name;
+	self.config = $.extend({}, defaultTrackConfig, config || {}); // inherit from default
+	
+	self.oncoprint = oncoprint;
+	self.data = data;
+
+	if (self.oncoprint.config.render === 'table') {
+		self.renderer = new TrackTableRenderer(self, new D3SVGCellRenderer(self));
+	}
+
+	self.getLabel = function() {
+		// TODO: label decorations
+		return self.config.label;
+	};
+	
+	self.getDatumIds = function(sort_cmp) {
```

Comment:
self.data is overwritten but not in a meaningful way - its order is modified; but its order is never used. But you're right it's not as clear and there's actually not even a need for that - that's a result of careless vestigial code writing.